modules/ebi-metagenomics/easel/eslsfetch/main.nf

```diff
@@ -0,0 +1,69 @@
+
+process EASEL_ESLSFETCH {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::easel=0.49"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/easel:0.49--h031d066_0':
+        'biocontainers/easel:0.49--h031d066_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    tuple val(meta), path(cmsearch_deoverlap_out)
+
+    output:
+    tuple val(meta), path("*easel_coords.fasta"), emit: easel_coords
+    tuple val(meta), path("*matched_seqs_with_coords"), emit: matched_seqs_with_coords
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def is_compressed = fasta.name.endsWith(".gz")
+    def fasta_name= fasta.name.replace(".gz", "")
+
```

Comment:
Switch to this, more Groovy and only checks extension as should:
def is_compressed = fasta.getExtension() == "gz" ? true : false
def fasta_name = is_compressed ? fasta.getBaseName() : fasta

Reply:
No need for the `? true : false` as `fasta.getExtension() == "gz"` boolean expression.